packages/react-celo/src/react-celo-reducer.ts

```diff
@@ -90,7 +97,14 @@ export interface ReducerState {
    */
   connectorInitError: Maybe<Error>;
   dapp: Dapp;
+  /**
+   * the chain id of the wallet (if applicable), may be different than dapp at certain moments,
+   * exposed here to give developer optionality in handling network behavior
+   * null when not known or wallet doesn't have a network (like ledger)
+   */
+  walletChainId: number | null;
```

Comment:
While I get it, I feel like this is semi-dangerous behaviour, let's make sure it's documented in the readme